src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -1659,6 +1659,11 @@ def calc_slope(data_sect, gdq_sect, frame_time, opt_res, save_opt, rn_sect,
     gdq_sect_r = np.reshape(gdq_sect, (ngroups, npix))
     mask_2d[gdq_sect_r != 0] = False  # saturated or CR-affected
     mask_2d_init = mask_2d.copy()  # initial flags for entire ramp
+    start = np.argmax(mask_2d, 0)  # start with the first True value
```

Comment:
It should be for everything. Given that this code is supposed to work for any instrument, it should do the same thing for all instruments.
The fix is for the case where the initial groups for a pixel are set to be skipped. If the initial group is True then start is zero. If the initial group is False start has to be the first good group. Otherwise, the bug will occur.
Today, only MIRI skips initial groups but nothing stops some future instrument for doing so.